src/Parser/TypeParser.php

```diff
@@ -53,6 +55,9 @@ private function parseAtomic(TokenIterator $tokens): Ast\Type\TypeNode
 
 			} elseif ($tokens->isCurrentTokenType(Lexer::TOKEN_OPEN_SQUARE_BRACKET)) {
 				$type = $this->tryParseArray($tokens, $type);
+
+			} elseif ($type->name === 'array' && $tokens->isCurrentTokenType(Lexer::TOKEN_OPEN_CURLY_BRACKET)) {
+				$type = $this->parseArrayShape($tokens, $type);
```

Comment:
this should be maybe `tryParseArrayShape`, so that `/** @return array {curly description} */` is not suddenly invalid

Reply:
The problem with `tryParse` is that it silently ignores errors. An array shape specification will be silently ignored if it has a minor typo in it.

Alternatively, if we don't allow spaces between `array` and `{`, this would prevent the problem you mention. We could do that by adding an `array{` token. What do you think ?

Reply:
> The problem with tryParse is that it silently ignores errors. 

That's not a problem of `tryParse` but inherent problem of phpDoc which is designed to be human friendly and not computer friendly (same problem as markdown).

Reply:
without `tryParse` it's a BC break.

Reply:
So what do you think of not allowing a whitespace between `array` and `{` ? This would prevent the BC break, while also avoiding silenced errors (which are not user friendly).

Reply:
Yes, that should be fine.

Reply:
https://github.com/phpstan/phpdoc-parser/pull/30/commits/03768b2e21f5c0953e394709fc0a018b259fba8b